autoload/ale/handlers/haskell.vim

```diff
@@ -72,6 +72,7 @@ function! ale#handlers#haskell#HandleGHCFormat(buffer, lines) abort
         \   'col': l:match[3] + 0,
         \   'text': l:text,
         \   'type': l:type,
+        \   'detail': join(a:lines, "\n"),
```

Comment:
That's very excessive detail. That adds _all_ of the lines of output to every item.

What you should do instead is add just some specific lines of output, and then cover your changes here with tests.